test/logflare/logs/logs_test.exs

```diff
@@ -187,4 +192,52 @@ defmodule Logflare.LogsTest do
       assert :ok = Logs.ingest_logs([%{"event_message" => "testing 123"}], source)
     end
   end
+
+  describe "System monitoring log" do
+    test "is routed to user's system source when monitoring is on", %{user: user, source: source} do
+      {:ok, user} = Users.update_user_allowed(user, %{system_monitoring: true})
+      system_source = Sources.get_by(user_id: user.id, system_source_type: :logs)
+
+      # Non-user-specific logs goes to the default logger backends
+      assert capture_log(fn -> Logger.info("common log") end) =~ "common log"
+
+      # User-specfic logs are routed to users with system monitoring on
+
+      refute capture_log(fn ->
+               Logger.info("user is monitoring", source_id: source.id)
+             end) =~ "user is monitoring"
+
+      assert Enum.any?(
+               Backends.list_recent_logs(system_source),
+               &match?(%{body: %{"event_message" => "user is monitoring"}}, &1)
+             )
+    end
+
+    test "is not routed to user's system source when not monitoring", %{
+      user: user,
+      source: source
+    } do
+      {:ok, user} = Users.update_user_allowed(user, %{system_monitoring: true})
+      Users.update_user_allowed(user, %{system_monitoring: false})
+      system_source = Sources.get_by(user_id: user.id, system_source_type: :logs)
+
+      assert capture_log(fn ->
+               Logger.info("user not monitoring", source_id: source.id)
+             end) =~ "user not monitoring"
+
+      refute Enum.any?(
+               Backends.list_recent_logs(system_source),
+               &match?(%{body: %{"event_message" => "user not monitoring"}}, &1)
+             )
+    end
+
+    setup do
+      :ok =
+        :logger.add_primary_filter(:user_log_intercetor, {&Logflare.UserLogInterceptor.run/2, []})
```

Comment:
Nice approach.
We should move the UserLogInterceptor under `Backends.Monitoring` namespace.